torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -7781,6 +7781,81 @@ def forward(_self, input, expected_type):  # noqa: B902
             inp = TestNamedTupleInput_1(a, b)
             model(inp, type(inp))
 
+        @require_backend_is_available({"gloo"})
+        def test_grads_same_across_ranks_with_no_sync(self):
+            group, group_id, rank = self._init_global_test()
+            world_size = dist.get_world_size()
+            if world_size < 2:
+                self.skipTest("This test requires at least two ranks.")
```

Comment:
nit: you can grep around this file to find decorators that skip with these sort of conditions.

Reply:
I did check, this exact condition is not met. There is a decorator with <= 2 `skip_if_small_worldsize`, but this test should be run with 2 ranks as well.